ci: "random versions" ci job failure for `lit_test`

This failed here: https://github.com/narwhals-dev/narwhals/actions/runs/21565401601/job/62135967097?pr=2989

```
=========================== short test summary info ============================
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[eager]-value2-dtype2] - TypeError: cannot create expression literal for value of type dict: {}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[lazy]-value2-dtype2] - TypeError: cannot create expression literal for value of type dict: {}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[eager]-value5-None] - TypeError: cannot create expression literal for value of type dict: {'field_1': 42}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[lazy]-value8-dtype8] - TypeError: cannot create expression literal for value of type dict: {'field_1': 42, 'field_2': 1.2, 'field_3': True}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[eager]-value8-dtype8] - TypeError: cannot create expression literal for value of type dict: {'field_1': 42, 'field_2': 1.2, 'field_3': True}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
FAILED tests/expr_and_series/lit_test.py::test_nested_structures[polars[lazy]-value5-None] - TypeError: cannot create expression literal for value of type dict: {'field_1': 42}

Hint: Pass `allow_object=True` to accept any value and create a literal of type Object.
```

versions used:
```
Using Python 3.9.25 environment at: /opt/hostedtoolcache/Python/3.9.25/x64
attrs==25.4.0
covdefaults==2.3.0
coverage==7.10.7
exceptiongroup==1.3.1
execnet==2.1.2
hypothesis==6.141.1
importlib-metadata==8.7.1
iniconfig==2.1.0
-e file:///home/runner/work/narwhals/narwhals
numpy==1.19.5
packaging==26.0
pandas==1.1.5
pip==26.0
pluggy==1.6.0
polars==1.0.0
pyarrow==13.0.0
pygments==2.19.2
pytest==8.4.2
pytest-cov==7.0.0
pytest-env==1.1.5
pytest-randomly==4.0.1
pytest-xdist==3.8.0
python-dateutil==2.9.0.post0
pytz==2025.2
setuptools==79.0.1
six==1.17.0
sortedcontainers==2.4.0
tomli==2.4.0
typing-extensions==4.15.0
zipp==3.23.0
```

--- a/tests/expr_and_series/lit_test.py
+++ b/tests/expr_and_series/lit_test.py
@@ -189,9 +189,9 @@ def test_nested_structures(
     if (
         "polars" in str(constructor)
         and isinstance(value, dict)
-        and POLARS_VERSION < (1, 0, 0)
+        and POLARS_VERSION < (1, 10, 0)
     ):  # pragma: no cover
-        reason = "polars<1.0 does not support dict to struct in lit"
+        reason = "polars<1.10 does not support dict to struct in lit"
         pytest.skip(reason=reason)
 
     size = 3

